snooty/postprocess.py

```diff
@@ -2143,6 +2143,47 @@ def exit_page(self, fileid_stack: FileIdStack, page: Page) -> None:
             page.ast.children.remove(node)
 
 
+class ComposableTutorialHandler(Handler):
+    """Handles composable tutorial directivepresence in page.
+    Should not be simultaneously present on page with other directives"""
+
+    def __init__(self, context: Context) -> None:
+        super().__init__(context)
+        self.target_directive_name = "composable-tutorial"
+        self.composable_tutorial = False
+        self.colliding_ast_options = ["selectors", "has_method_selector"]
+        self.composable_node_start = 0
+
+    def enter_page(self, fileid_stack: FileIdStack, page: Page) -> None:
+        self.composable_tutorial = False
+
+    def enter_node(self, fileid_stack: FileIdStack, node: n.Node) -> None:
+        if not isinstance(node, n.ComposableDirective):
+            return
+
+        if self.composable_tutorial:
+            self.context.diagnostics[fileid_stack.current].append(
+                DuplicateDirective(self.target_directive_name, (node.start[0]))
+            )
+            return
+
+        self.composable_tutorial = True
+        self.composable_node_start = node.start[0]
+
+    def exit_page(self, fileid_stack: FileIdStack, page: Page) -> None:
+        if not self.composable_tutorial:
+            return
+        for colliding_ast_option in self.colliding_ast_options:
+            if page.ast.options.get(colliding_ast_option, None):
+                self.context.diagnostics[fileid_stack.current].append(
+                    UnexpectedDirectiveOrder(
+                        f"{self.target_directive_name} cannot be used with {colliding_ast_option} on the same page",
```

Comment:
(Nit) I think `["selectors", "has_method_selector"]` are options specific to internal implementation and are not user facing. To make this error more user friendly, would it make sense to print out the associated directive in the error instead (`tabs-selector` and `method-selector`)?

Reply:
updated! this is a bit hardcoded at this handler as a tuple, but feel free if you have other suggestions here

Reply:
I think this is fine for now since it's a pretty small list.

If the list grows/changes in the future, we can maybe have the `enter_node` method look for and keep track if one of those directives were seen on the same page.